nsv13/code/modules/overmap/weapons/plasma_gun.dm

```diff
@@ -0,0 +1,466 @@
+	flick("[initial(icon_state)]_firing",src)
+	sleep(fire_animation_length)
+	icon_state = initial(icon_state)
+
+/obj/machinery/ship_weapon/plasma_caster/animate_projectile(atom/target)
+	return linked.fire_projectile(weapon_type.default_projectile_type, target, homing = TRUE, speed = 2, lateral=weapon_type.lateral)
+
+/obj/machinery/ship_weapon/plasma_caster/proc/misfire()
+	say("WARNING! Phoron containment field failure, ejecting gas!")
+	if(prob(15))
+		do_sparks(4, FALSE, src)
+	if(plasma_mole_amount > 0 && prob(5))
+		makedarkpurpleslime()
+	atmos_spawn_air("plasma=[plasma_mole_amount];TEMP=293")
+	alignment -= rand(30,90)
+	if(alignment < 0)
+		alignment = 0
+	field_integrity -= rand(20,50)
+	plasma_mole_amount = 0
+	playsound(src, malfunction_sound, 100, 1)
+	tesla_zap(src, 8, TESLA_MINI_POWER, TESLA_ENERGY_MINI_BALL_FLAGS)
+	playsound(src,'sound/machines/defib_zap.ogg', 100, 1)
+	unchamber()
+	unload()
+
+/obj/machinery/ship_weapon/plasma_caster/overmap_fire(atom/target)
+
+	if(weapon_type?.overmap_firing_sounds)
+		overmap_sound()
+
+	if(overlay)
+		overlay.do_animation()
+	sleep(4 SECONDS)
+	if( weapon_type )
+		animate_projectile(target)
+
+/obj/machinery/ship_weapon/plasma_caster/after_fire()
+	alignment -= rand(30,60)
+	if(alignment < 0)
+		alignment = 0
+	plasma_mole_amount = 0
+	field_integrity -= 20
+	cooldown = 100
+	light_power = 2
+	..()
+
+/obj/machinery/ship_weapon/plasma_caster/default_deconstruction_crowbar(obj/item/I, ignore_panel)
+	var/mob/living/fool = usr
+	var/confirm = alert("You feel an electric tingle as you bring your crowbar close, there's an odd heat the closer you get.\
+						 Are you sure this is a good idea?", "Deconstruct Phoron Caster", "Yes", "No")
+	if(confirm == "Yes")
+		visible_message("<span class='danger'>Burning energy and phoron starts to vent from the gun which chars [usr]!</span>")
+		field_integrity -= rand(30,50)
+		alignment -= rand(50,80)
+		if(alignment < 0)
+			alignment = 0
+		fool.adjustFireLoss(rand(50, 120)) // Don't try to deconstruct it
+		fool.IgniteMob()
+		playsound(usr.loc, 'sound/magic/lightningbolt.ogg', 100, 1, extrarange = 30)
+		return
+	else
+		visible_message("<span class 'notice'>[usr] wisely lowers their crowbar.")
+		return
+
+/obj/machinery/ship_weapon/plasma_caster/multitool_act(mob/living/user, obj/item/I)
+	. = TRUE
+	if(maint_state == 0)
+		to_chat(user, "<span class='notice'>You must first open the maintenance panel before unwrenching the protective casing!</span>")
+	if(maint_state == 1)
+		to_chat(user, "<span class='notice'>You must unbolt the protective casing before tuning the magnetic frequency!</span>")
+	else
+		to_chat(user, "<span class='notice'>You being tuning the magnetic frequency.</span>")
+		while(alignment < 100)
+			if(!do_after(user, 5, target = src))
+				return
+			alignment += rand(1,2)
+			if(alignment >= 100)
+				alignment = 100
+				break
+
+/obj/machinery/ship_weapon/plasma_caster/attack_hand(mob/user)
+	ui_interact(user)
+	return
```

Comment:
```suggestion
```
This is already handled by machinery/attack_hand(), it also lets you unload rounds that are not chambered yet (though maybe that is why you overrode it)

Reply:
You unload by clicking the load button in the UI, I don't want someone to do with their bare hands.